public/mobile-app/src/lib/notifications.ts

```diff
@@ -107,7 +97,38 @@ export const enableNotifications = async () => {
       'No notification: missing permission or missing service worker registration'
     )
   } else {
-    await registerUser()
+    const pushSubscription = await subscribePush()
+    return await registerDevice(pushSubscription)
+  }
+}
+
+export const unsubscribePush = async (pushSubscription) => {
+  try {
+    const hasUnsubscribed = await pushSubscription.unsubscribe()
+    if (hasUnsubscribed) {
+      console.log('unsubscribed to the push manager')
+    } else {
+      console.log('failed to unsubscribe to the push manager')
+    }
+  } catch (error) {
+    console.error(error)
+  }
+}
```

Comment:
How about returning a boolean (if it unsubscribed properly => `true`), and letting the caller deal with the hypothetical error?
This would make it more easily testable, and would leave the responsability to the caller to decide what to do with the error (instead of just logging it).

What do you think?